modules/040-node-manager/hooks/handle_draining.go

```diff
@@ -189,6 +208,48 @@ func handleDraining(input *go_hook.HookInput, dc dependency.Container) error {
 	return nil
 }
 
+func getDrainTimeout(input *go_hook.HookInput, client k8s.Client, ngName string) time.Duration {
+	nodeGroups, err := client.Dynamic().Resource(nodeGroupResource).Namespace("").List(context.TODO(), v1.ListOptions{})
```

Comment:
first, we can add cache for it, because we can drain multiple nodes in one nodegroup, and we should not request to api multiple times only one 1

Reply:
added drainTimeoutCache := make(map[string]time.Duration)